ibis/expr/datatypes.py

```diff
@@ -1516,10 +1566,15 @@ def can_cast_variadic(
 
 
 # geo spatial data type
+# cast between same type, used to cast from/to geometry and geography
 @castable.register(Array, Point)
 @castable.register(Array, LineString)
 @castable.register(Array, Polygon)
 @castable.register(Array, MultiPolygon)
+@castable.register(Point, GeoSpatial)
+@castable.register(LineString, GeoSpatial)
+@castable.register(Polygon, GeoSpatial)
+@castable.register(MultiPolygon, GeoSpatial)
```

Comment:
You can make this a bit more succinct by using `@castable.register((Point, LineString, Polygon, MultiPolygon), GeoSpatial)`. Same with the `Array` above though that isn't necessary for this PR.

Reply:
sounds much better. I will do it! thanks!